### ISSUE TITLE
SuppressWarnings: no support for module declarations

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#21163]
Parent Issue #21113

I have read check documentation: https://checkstyle.org/checks/annotation/suppresswarnings.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Note: module-info.java is only parseable after #8240, so this is reproduced on that branch.

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="SuppressWarnings"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > sw/module-info.java <<'EOF'
@SuppressWarnings("")
module com.example.app {
    requires java.base;
}
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.10.0-SNAPSHOT-all.jar -c config.xml sw/module-info.java

Starting audit...
Audit done.
```

The same annotation on a class is reported:

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > A.java <<'EOF'
package com.example;

@SuppressWarnings("")
class A {
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java $RUN_LOCALE -jar target/checkstyle-13.10.0-SNAPSHOT-all.jar -c config.xml A.java

Starting audit...
[ERROR] A.java:3:19: The warning '' cannot be suppressed at this location. [SuppressWarnings]
Audit done.
Checkstyle ends with 1 errors.
```

**Describe what you expect in detail.**

`@SuppressWarnings` on a module declaration is legal, javac compiles the module input above without error, so the Check should validate it the same way it validates the annotation on a class.

The Check's acceptable tokens are thirteen declaration types (`CLASS_DEF`, `INTERFACE_DEF`, `ENUM_DEF`, `ANNOTATION_DEF`, `ANNOTATION_FIELD_DEF`, `ENUM_CONSTANT_DEF`, `PARAMETER_DEF`, `VARIABLE_DEF`, `METHOD_DEF`, `CTOR_DEF`, `COMPACT_CTOR_DEF`, `RECORD_DEF`, `PATTERN_VARIABLE_DEF`) with no `MODULE_DEF`, so a module declaration is never visited.

Expected on the module input:

```
1:19: The warning '' cannot be suppressed at this location.
```

*New:*
New acceptable token `TokenTypes.MODULE_DEF`

**Comment by romani** [comment]
Fix is merged
